src/hooks/useTabs.tsx

```diff
@@ -0,0 +1,60 @@
+"use client";
+import { createContext, useContext, useState } from 'react';
+import { LinkProps } from 'next/link';
+import { usePathname } from 'next/navigation';
+import { explorerFiles } from '@/components/Explorer';
+
+type TabsType = {
+  children: React.ReactNode[];
```

Comment:
I don't think we should store the data here as ReactNode, we should instead probably only store a reference to the open file, not all of its props.

We can have somewhere a file with the list of all files we have in the explorer (we already have that at `components/Explorer/index.tsx`), something like:

```js
export const explorerFiles = {
  '/vscode/general': {
    icon: <SomeIcon />
    title: 'General',
    // any other prop
  }
}
```

And inside the context, when we open a file, we should save only the file reference, something like:

```js
const openFiles = ['/vscode/general', '/vscode/settings']
```

What do you think?